### ISSUE TITLE
batch_size problem

### ISSUE THREAD

**Issue by LeZhengThu** [mexchy1000/DeepSurv_Keras#1]
Hi, when I change the batch_size parameter from 324 to any other integer, say 103, keras threw out an error said 'ValueError: could not broadcast input array from shape (1,324) into shape (1,103)'. Do you have any idea what this means and how can this be corrected? Thanks.

**Comment by LeZhengThu** [comment]
I thought it was a loss function problem. In line 70, you defined 'loss=negative_log_likelihood(E_train)' in model.compile, and the E_train has limited the size of the input data. How can we modify this to let the code can be trained in a mini-batch manner?

**Comment by mexchy1000** [comment]
@LeZhengThu Loss function for survival data cannot be obtained by a mini-batch manner. That's because loss of a subject is calculated by sum of log-likelihood of censored patients except for the subject,

**Comment by LeZhengThu** [comment]
@mexchy1000 I changed the output to contains survived time and labels simultaneously. Then the loss function can be obtained in a mini-batch manner because the censored information are included in each batch.

**Comment by mexchy1000** [comment]
@LeZhengThu Thanks! Can you show your code or make a new branch? Did you sort the data for each batch according to the survival time and trained the network using train_on_batch function? I think this is a good solution, but I'm afraid the mini-batch manner may produce inappropriate hazard ratio as HR is a relative value of all survival data, not dependent on batch.

**Comment by LeZhengThu** [comment]
@mexchy1000 Yes, the inappropriate hazard ratio is a problem. Can you help check if my modification is correct. The new loss function is like:

def sur_loss(y_true,y_pred):
        hazard_ratio = T.exp(y_pred[:,0])
        log_risk = T.log(T.extra_ops.cumsum(hazard_ratio))
        uncensored_likelihood = y_pred[:,0].T - log_risk
        censored_likelihood = uncensored_likelihood * y_true[:,1]
        neg_likelihood = -T.sum(censored_likelihood)
        return neg_likelihood

Where y[:,0] is survived time and y[:,1] is label.

**Comment by LeZhengThu** [comment]
@mexchy1000 Hi, in the code line 39 why do you do the sorting? And in line 73 why setting shuffle=False is important?

**Comment by IsDling** [comment]
@LeZhengThu Can you get good result with mini_batch? I train my model with mini_batch but it maybe not work.
